### DISCUSSION TITLE
Is there any more documentation on implementing teams?

### DISCUSSION THREAD

**Discussion by DanielTate** [spatie/laravel-permission#1998] [Q&A] [answered]
Hey I have followed your guides and have at least two models in my app.
Users and Accounts ( Accounts is equivalent to Teams ).

I have extended both of these with the HasRoles Trait that laravel-permissions provides.

I have updated my permission.php to include account_id as the custom team_foreign_key

I have created seeders the create users accounts and roles. The users accounts and roles are all seeded correctly and show up in the database.
users
|id |name       |email            |
|---|-----------|-----------------|
|1  |Super Admin|admin@example.com|

accounts
|id |name       |email            |
|---|-----------|-----------------|
|1  |Dummy Account|bryana.collier@abshire.com|

account_user
|id |account_id |user_id          |
|---|-----------|-----------------|
|1  |1          |1                |

roles
|id |account_id |name             |guard_name|
|---|-----------|-----------------|----------|
|1  |1          |Super Admin      |web       |

model_has_roles
|role_id|model_type |model_id         |account_id|
|-------|-----------|-----------------|----------|
|1      |App\Models\User|1                |1         |

I implemented some middleware and put it in the priority list although I'm not sure where it should be, should it be first or last? Should I add the other spatie permission classes to this priority list also?

Everything looks like it's hooked up correctly following the Teams guide however when I do the following in my middleware:

`setPermissionsTeamId(1)`

`getPermissionsTeamId() // null`

The result is always null. If I set and get from a policy or from a controller.

I checked the query for roles and it seems that the team isn't being applied. Have I missed something?

```Role::first()->toSql() // "select * from `roles`"```

Shouldn't this be ```"select * from `roles` where `account_id` = 1"```

This part of the documentation https://spatie.be/docs/laravel-permission/v5/basic-usage/teams-permissions#working-with-teams-permissions was particularly confusing. I can't figure out what that sentence means. Any help is appreciated. 

Even some working examples would be nice to look at.

**Comment by erikn69** [comment, marked as answer]
Version of package?
https://github.com/spatie/laravel-permission/discussions/1906#discussioncomment-1584335

**Reply by DanielTate** [reply]
5.4.0

I have done some more testing and it's definitely a fault in my application. It seems like there's something I've done preventing the setPermissionsTeamId method from working correctly. Probably leave this open until I find the specific issue and can post an answer that may help someone else.

**Reply by DanielTate** [reply]
So I couldn't find anything wrong in my application. I literally turned everything off but upgrading to 5.5.0 seemed to fix the problem. Either the install went wrong or there was a change in 5.5.0 that fixed everything. I would recommend anyone else to reinstall or upgrade from 5.4.0 to 5.5.0 since the API is the same.
